### ISSUE TITLE
Remove compiler.jar from repository and resolve JS style problems with newer version

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#1761]
Google moved/removed some old versions of their Closure compiler tools from their old download areas, breaking testing. For the OJS 3.0 release I moved my copy into the repo, which is not a great solution.

Remove it, and get things going with the latest stable version of the compiler.

(See: https://github.com/pkp/pkp-lib/commit/12c7be362bfb76c18302116b5f5a4cee86156794)

**Comment by asmecher** [comment]
Cherry-picked https://github.com/pkp/pkp-lib/commit/12c7be362bfb76c18302116b5f5a4cee86156794 to `omp-stable-1_2_0`, `omp-dev-1_2`, `ojs-stable-2_4_8`, and `ojs-dev-2_4`. That copy of the compiler jar will probably be the permanent solution for those branches; we'll modernize our JS stuff in master for the future and likely not back-port it.

**Comment by asmecher** [comment]
@NateWr, this will be a good candidate after the merge of your JS work, but not a high priority for 3.1.

**Comment by NateWr** [comment]
@asmecher, I know you've already got enough on your plate, but if you get a chance you should look into [Webpack](https://webpack.js.org/). 

We use it to compile our `.vue` files into a JS bundle. But it's really a more generic file processor that could be used to automate the whole build and packaging process.

**Comment by NateWr** [comment]
@asmecher presumably this is outdated and no longer relevant? Reopen if I made a mistake.

**Comment by asmecher** [comment]
I'm content to not touch this until we have to. It'll be a long time before we can remove the old JS toolkit but at least it's stable and low-risk.
